.github/CONTRIBUTING.md

```diff
@@ -755,7 +754,288 @@ It can always be reopened if it was a mistake.
+
+The following points can help you improve the readability of your tests:
+
+* Make proper spacing between `it` blocks 
+* Group related tests under common `describe` blocks 
+* Keep your tests isolated (more on this here: [The good test is isolated and flat](#the-good-test-is-isolated-and-flat))
+* Follow the AAA pattern as described above
+
+Below there are two examples. Which one is easiest to read? 
+
+![](images/unreadable-tests.png)
+
+![](images/readable-tests.png)
+
+(Our guess is the second one)
+
+#### The good test is isolated and flat
+
+The tests should be isolated and flat, therefore writing tests is one of the few places, where it is okay to be [WET](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself#DRY_vs_WET_vs_AHA_solutions) instead of [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) - it is rather important actually. 
+
+Future contributors should be able to read the test without having to jump around the code and do too much logic, therefore: 
+
+* Avoid reusing code by using functions, keep your tests flat instead 
+* Avoid the use of shared variables between tests, all state should be isolated to the test 
+* Use `beforeEach` instead of `beforeAll`; it will cause potential changes made by previous tests to be overwritten, and you start with the same state each time
+
+Following the above points makes it more likely that tests fail due to actual code changes and not how the tests are written. Changing a shared function can cause several tests to fail/change unexpectedly. If it happens too often that the test is the problem, not the code; it can cause mistrust in these.
+
+Therefore be WET - it will lower the chance of the tests being the problem and give a better overview of what is going on when reading the test. 
+
+When doing unit tests, further isolation has to be done by stubbing and mocking everything else than what is being tested. 
+
+#### The good test uses test scenarios when appropiate 
+
+If you find yourself writing several tests that have identical arrange, act and assess sections - just with different variables, then you might benefit from using test scenarios. 
+
+An array of scenarios are used to programatically create tests for each using the `forEach` function. Each test scenario contains the variables making up the scenario and then the expected outcome of this. 
+
+Take for example the [ `kirby-button` ](https://cookbook.kirby.design/home/showcase/button) directive which is used to render buttons. Depending on which size is passed as an input property, it is rendered with different values for `font-size` , `height` and `min-width` . When testing this, the tests for each size is identical. They only differ in which value is given for size and the expected outcome. This can be expressed as an array of test scenarios: 
+
+``` 
+
+const testScenarios: { size: ButtonSize; expected: any }[] = [
```

Comment:
Consider either strong typing `expected` or removing the type altogether for simplicity of the example: 
```suggestion
const testScenarios = [
```

Reply:
Removed it for simplicities sake